### PR TITLE
refactor consume_to_file_ordered

### PR DESCRIPTION
**Pull request by garrettthomaskth**

### CODE REVIEW

File: esque/clients/consumer.py [hunk 1]
```diff
@@ -295,13 +298,34 @@ def consume_to_file_ordered(
     output_directory: pathlib.Path,
     topic: str,
     group_id: str,
-    partitions: list,
-    numbers: int,
+    partitions: List[int],
+    desired_count_messages: int,
```
Comment (swenzel) [inline esque/clients/consumer.py]: `desired_message_count` would sound a little more natural

File: esque/clients/consumer.py [hunk 2]
```diff
@@ -316,63 +340,68 @@ def consume_to_file_ordered(
...
-                    partitions.remove(partition_counter)
-                    if len(partitions) == 0:
-                        messages_left = False
+                # a possible timeout due to a network issue, retry (but not more than MAX_RETRY_COUNT attempts)
+                # retry_count = _handle_empty_message_exception(retry_count)
+                retry_count = retry_count + 1
+                if retry_count > MAX_RETRY_COUNT:
                     keep_polling_current_partition = False
+                    logger.debug(
+                        f"Could not connect to {partition_counter}. "
```
Comment (swenzel) [inline esque/clients/consumer.py]:
```suggestion
                        f"Could not read from partition {partition_counter}. "
```

File: esque/cli/commands.py [hunk 3]
```diff
@@ -629,7 +629,7 @@ def get_topics(state: State, prefix: str, output_format: str):
     required=False,
 )
 @click.option(
-    "-n", "--numbers", metavar="<n>", help="Number of messages.", type=click.INT, default=sys.maxsize, required=False
+    "-n", "--number", metavar="<n>", help="Number of messages.", type=click.INT, default=sys.maxsize, required=False
```
Comment (swenzel) [inline esque/cli/commands.py:632]: This is a breaking change. I think we should keep `--numbers` around for compatibility reasons and emit a deprecation warning.

Comment (garrettthomaskth) [reply esque/cli/commands.py:632]: So we agreed to just release a new version with breaking changes, right? CC: @hfjn

Comment (hfjn) [reply esque/cli/commands.py:632]: if you bump the version number accordingly go for it.

**Review by swenzel — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+### Version 0.2.2
+* Consume option renamed from `numbers` to `number`.
+* Add `edit offsets` command.
+* Bugfix for `OverflowError` in `pretty_duration` function.
```
Comment (swenzel) [inline CHANGELOG.md:4]:
```suggestion
* Bugfix for `OverflowError` in `pretty_duration` function.
* use `esque-client` as default group id for all esque commands
* disable group offset commits for ping, consume and describe
```

**Review by swenzel — APPROVED** [review]

**Pull request merged**